packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,430 @@
+
+    const jsonLdCredential = new JsonLdCredential(requestData)
+    MessageValidator.validateSync(jsonLdCredential)
+
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: JSONLD_VC_DETAIL,
+    })
+
+    const attachment = this.getFormatData(requestData, format.attachId)
+    return { format, attachment }
+  }
+
+  /**
+   * Create a credential attachment format for a credential request.
+   *
+   * @param options The object containing all the options for the credential request is derived
+   * @returns object containing associated attachment, formats and requestAttach elements
+   *
+   */
+  public async createRequest(
+    agentContext: AgentContext,
+    { credentialFormats }: FormatCreateRequestOptions<JsonLdCredentialFormat>
+  ): Promise<CredentialFormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    const format = new CredentialFormatSpec({
+      format: JSONLD_VC_DETAIL,
+    })
+
+    if (!jsonLdFormat) {
+      throw new AriesFrameworkError('Missing jsonld payload in createRequest')
+    }
+
+    const jsonLdCredential = new JsonLdCredential(jsonLdFormat)
+    MessageValidator.validateSync(jsonLdCredential)
+
+    const attachment = this.getFormatData(jsonLdFormat, format.attachId)
+
+    return { format, attachment }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: FormatProcessOptions): Promise<void> {
+    const requestJson = attachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    if (!requestJson) {
+      throw new AriesFrameworkError('Missing jsonld credential request data payload')
+    }
+
+    const jsonLdCredential = new JsonLdCredential(requestJson)
+    MessageValidator.validateSync(jsonLdCredential)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { credentialFormats, attachId, requestAttachment }: FormatAcceptRequestOptions<JsonLdCredentialFormat>
+  ): Promise<CredentialFormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    // sign credential here. credential to be signed is received as the request attachment
+    // (attachment in the request message from holder to issuer)
+    const credentialRequest = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    const credentialData = jsonLdFormat ?? credentialRequest
+    const jsonLdCredential = new JsonLdCredential(credentialData)
+    MessageValidator.validateSync(jsonLdCredential)
+
+    const verificationMethod =
+      credentialFormats?.jsonld?.verificationMethod ??
+      (await this.deriveVerificationMethod(agentContext, credentialData.credential, credentialRequest))
+
+    if (!verificationMethod) {
+      throw new AriesFrameworkError('Missing verification method in credential data')
+    }
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: JSONLD_VC,
+    })
+
+    const verifiableCredential = await this.w3cCredentialService.signCredential(agentContext, {
```

Comment:
We're ignoring some properties here (domain, challenge, credentialStatus) from the `credentialData.options` object here. Ideally we support them, but we can add that in a later PR also. What is important is to throw an error when any of those fields are present in the options that we don't support them. Because this can lead to issuance of credentials that don't match what is negotiated. Having a challenge can be an important aspect of preventing replay-attacks so we need to make sure we're throwing errors if we're not following what the exchagne discussed.

Reply:
> > This is urgent (to be merged) as it costs @NB-MikeRichardson a lot of overhead.
> 
> Yes this has been outstanding for way too long. Sorry on this from my side for reviewing it slowly. I think as discussed before: smaller PRs that are easier to review. As a tip to you in the future Mike, please read the RFC carefully and handle all edge cases before making a PR. This will save me a lot of time in reviewing, and save you a lot of time in doing multiple round trips.

No need to apologise; I appreciate the amount of time you've put into this and I know how busy you are.. And it helps me become a better Aries/AFJ/Typescript developer.

Reply:
Moved this into Issue #1089 for separate PR

https://github.com/hyperledger/aries-framework-javascript/issues/1089